### PR TITLE
Possible fix for 2069

### PR DESCRIPTION
**Pull request by sethalves**
- use shared-pointers for EntityTreeRenderer

### CODE REVIEW

**Comment by sethalves** [comment]
https://highfidelity.fogbugz.com/f/cases/2069/

**Comment by sethalves** [comment]
to test:
- no functional changes

File: interface/src/octree/OctreePacketProcessor.cpp [hunk 1]
```diff
@@ -92,13 +92,19 @@ void OctreePacketProcessor::processPacket(QSharedPointer<ReceivedMessage> messag
     switch(packetType) {
         case PacketType::EntityErase: {
             if (DependencyManager::get<SceneScriptingInterface>()->shouldRenderEntities()) {
-                qApp->getEntities()->processEraseMessage(*message, sendingNode);
+                auto renderer = qApp->getEntities();
+                if (renderer) {
+                    renderer->processEraseMessage(*message, sendingNode);
+                }
```
Comment (sethalves) [inline interface/src/octree/OctreePacketProcessor.cpp]: Here is the actual fix for the crash -- I think getEntities can return a null pointer during shutdown.

**Comment by hifi-gustavo** [comment]
Build console output for commit: f47aeae2a74cd63b8fc5347e0eaa39010a62557b available at: https://gist.github.com/b54837606fe43898cde1808fdabda2d4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8931-f47aeae.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8931-f47aeae.dmg

**Comment by ZappoMan** [comment]
Build failed... why use `QSharedPointer<>` and not an `stl::shared_ptr<>`? otherwise... sure... I approve.

**Comment by sethalves** [comment]
Because DependencyManager uses QSharedPointers.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ea3b513e71b4b556f16d32498b4f8a5f7fa0e0f1 available at: https://gist.github.com/b54837606fe43898cde1808fdabda2d4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8931-ea3b513.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8931-ea3b513.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: edb3faac789a62b93d9d94fa3779b11ba9cf5b69 available at: https://gist.github.com/b54837606fe43898cde1808fdabda2d4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8931-edb3faa.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8931-edb3faa.dmg

**Comment by cozza13** [comment]
- Mouse does not work. Try moving some blocks with the mouse. 
- Crashed on exit

**Comment by sethalves** [comment]
There's something else up.  On my windows machine, I can move things with the mouse.  On my Linux machine, which is a build of master, I can't.

**Comment by sethalves** [comment]
The problems that caused Chris to QA fail this were in master.  Please re-QA

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3ad748cfe227be31aa5db679b250192b3832df6a available at: https://gist.github.com/b54837606fe43898cde1808fdabda2d4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8931-3ad748c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8931-3ad748c.dmg

**Comment by cozza13** [comment]
Tested and approved

**Pull request merged**